src/main/java/org/terasology/gooeyDefence/EnemyManager.java

```diff
@@ -85,6 +108,39 @@ public void onPeriodicActionTriggered(PeriodicActionTriggeredEvent event, Entity
         }
     }
 
+    /**
+     * Called when the a path is changed.
+     *
+     * @see OnPathChanged
+     */
+    @ReceiveEvent
+    public void onPathChanged(OnPathChanged event, EntityRef shrineEntity) {
+        for (EntityRef enemy : enemies) {
+            /* Firstly check if the enemy is on an unchanged path */
+            if (enemy.hasComponent(EntrancePathComponent.class)) {
+                if (enemy.getComponent(EntrancePathComponent.class).getEntranceId() != event.getPathId()) {
+                    return;
+                }
+            }
+
+            /* Check if the goal is on the new path */
+            PathComponent pathComponent = getPathComponent(enemy);
+            Vector3i goal = pathComponent.getGoal();
+            List<Vector3i> newPath = event.getNewPath();
+            if (newPath.contains(goal)) {
+                /* Add a entrance component starting at the given position */
+                enemy.removeComponent(pathComponent.getClass());
+                EntrancePathComponent entranceComponent = new EntrancePathComponent(
+                        event.getPathId(),
+                        pathfindingSystem,
+                        newPath.indexOf(goal));
+                enemy.addComponent(entranceComponent);
+            } else {
+                /* It's had its path change and it isn't on the new path */
+                enemy.send(new RepathEnemyRequest());
```

Comment:
ok so it took me a while to get it but I guess by OnPathChanged you mean OnPathToEntranceChanged. When you have  multiple components for Paths, it's a definite source of complexity and confusion. So I think names should be as explicit as possible

Reply:
This has been fixed.